builtin/providers/aws/resource_aws_ami.go

```diff
@@ -196,7 +196,9 @@ func resourceAwsAmiRead(d *schema.ResourceData, meta interface{}) error {
 	d.Set("ebs_block_device", ebsBlockDevs)
 	d.Set("ephemeral_block_device", ephemeralBlockDevs)
 
-	d.Set("tags", tagsToMap(image.Tags))
+	if _, ok := d.GetOk("tags"); ok {
+		d.Set("tags", tagsToMap(image.Tags))
```

Comment:
Any thoughts for working around this? We could give resources two attributes, `tags` which is a map of strings to strings like before, and `tags_managed_by_attachment` which is a boolean. Then instead of `_, ok := d.GetOk("tags"); ok` it would be `!d.Get("tags_managed_by_attachment").(bool)`.